soc/arm/microchip_mec/common/soc_i2c.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/sys_io.h>
+#include <sys/__assert.h>
+#include <soc.h>
+#include "soc_i2c.h"
+
+#define MCHP_XEC_GPIO_REG_BASE	((struct gpio_regs *)0x40081000u)
```

Comment:
this information is already in DT, why is this needed? You can use DT API to obtain this one.

Reply:
using DT API